File: react-test-app/src/services/user-service.ts
import axios from "axios";
import type { User, UserRequest } from "../types/user";

const API_BASE_URL = "/api/users";

export const userService = {
  // 全ユーザー取得
  async getAllUsers(): Promise<User[]> {
    const response = await axios.get<User[]>(API_BASE_URL);
    return response.data;
  },

  // IDでユーザー取得
  async getUserById(id: number): Promise<User> {
    const response = await axios.get<User>(`${API_BASE_URL}/${id}`);
    return response.data;
  },

  // メールアドレスでユーザー取得
  async getUserByEmail(email: string): Promise<User> {
    const response = await axios.get<User>(`${API_BASE_URL}/email/${email}`);
    return response.data;
  },

  // 名前で検索
  async searchUsersByName(name: string): Promise<User[]> {
    const response = await axios.get<User[]>(`${API_BASE_URL}/search`, {
      params: { name },
    });
    return response.data;
  },

  // ユーザー作成
  async createUser(userRequest: UserRequest): Promise<User> {
    const response = await axios.post<User>(API_BASE_URL, userRequest);
    return response.data;
  },

  // ユーザー更新
  async updateUser(id: number, userRequest: UserRequest): Promise<User> {
    const response = await axios.put<User>(
      `${API_BASE_URL}/${id}`,
      userRequest
    );
    return response.data;
  },

  // ユーザー削除
  async deleteUser(id: number): Promise<void> {
    await axios.delete(`${API_BASE_URL}/${id}`);
  },
};

File: demo/src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import com.example.demo.entity.User;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class UserService {
    
    @Autowired
    private UserRepository userRepository;
    
    // 全ユーザー取得
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }
    
    // IDでユーザー取得
    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }
    
    // メールアドレスでユーザー取得
    public Optional<User> getUserByEmail(String email) {
        return userRepository.findByEmail(email);
    }
    
    // 名前で検索
    public List<User> searchUsersByName(String name) {
        return userRepository.findByNameContaining(name);
    }
    
    // ユーザー作成
    public User createUser(String name, String email) {
        User user = new User(name, email);
        return userRepository.save(user);
    }
    
    // ユーザー更新
    public Optional<User> updateUser(Long id, String name, String email) {
        Optional<User> userOpt = userRepository.findById(id);
        if (userOpt.isPresent()) {
            User user = userOpt.get();
            user.setName(name);
            user.setEmail(email);
            return Optional.of(userRepository.save(user));
        }
        return Optional.empty();
    }
    
    // ユーザー削除
    public boolean deleteUser(Long id) {
        if (userRepository.existsById(id)) {
            userRepository.deleteById(id);
            return true;
        }
        return false;
    }
}